app/models/attachment.js

```diff
@@ -63,43 +71,114 @@ exports.addModel = function(database) {
         })
-        .then(function(res) { return Post.findById(that.postId) })
-        .then(function(post) { return post.addAttachment(that.id)})
         .then(function(res) { resolve(that) })
         .catch(function(e) { reject(e) })
     })
   }
 
-  Attachment.prototype.destroy = function() {
+  // Get user who created the attachment (via Promise, for serializer)
+  Attachment.prototype.getCreatedBy = function() {
+    return models.FeedFactory.findById(this.userId)
+  }
+
+  // Get public URL of attachment (via Promise, for serializer)
+  Attachment.prototype.getUrl = function() {
     var that = this
+    return new Promise(function(resolve, reject) {
+      resolve(config.attachments.urlDir + that.getFilename())
+    })
+  }
 
+  // Get public URL of attachment's thumbnail (via Promise, for serializer)
+  Attachment.prototype.getThumbnailUrl = function() {
+    var that = this
     return new Promise(function(resolve, reject) {
-      database.delAsync(mkKey(['attachment:', that.id]))
-        .then(function(res) {
-          return Promise.all([
-            database.publishAsync('destroyAttachment',
-                                  JSON.stringify({
-                                    postId: that.postId,
-                                    attachmentId: that.id
-                                  })),
-            database.delAsync(mkKey(['attachment', that.id])),
-            database.lremAsync(mkKey(['post', that.postId, 'attachments']), 1, that.id)
-          ])
-        })
-        .then(function(res) { resolve(res) })
+      if (that.noThumbnail) {
+        resolve(that.getUrl())
+      } else {
+        resolve(config.attachments.thumbnails.urlDir + that.getFilename())
+      }
+    })
+  }
+
+  // Get local filesystem path for original file
+  Attachment.prototype.getPath = function() {
+    return config.attachments.fsDir + this.getFilename()
+  }
+
+  // Get local filesystem path for thumbnail file
+  Attachment.prototype.getThumbnailPath = function() {
+    return config.attachments.thumbnails.fsDir + this.getFilename()
+  }
+
+  // Get file name
+  Attachment.prototype.getFilename = function() {
+    if (this.fileExtension) {
+      return this.id + '.' + this.fileExtension
+    }
+    return this.id
+  }
+
+  // Rename file and process thumbnail, if necessary
+  Attachment.prototype.handleMedia = function(attachment) {
+    var that = this
+
+    return new Promise(function(resolve, reject) {
+      var tmpPath = that.file.path
+      var originalPath = that.getPath()
+
+      fs.rename(tmpPath, originalPath, function() {
+        if ('image') { // TODO: support for various media types
+          gm(originalPath).size(function (err, size) {
+            // Check if we need to resize
+            if (size !== undefined && (size.width > 525 || size.height > 175)) {
+              // Looks big enough, needs a resize
+              that.noThumbnail = ''
```

Comment:
Well, the reason I chose `'true'` and `''` for this, is using `if (that.noThumbnail)` as it's a usual boolean. If we change them to `'1'` and `'0'`, we should explicitly use `if (that.noThumbnail === '1')` instead, which is quite ugly.